### PR TITLE
etcd: Update to version 3.5.7

### PR DESCRIPTION
**Pull request by mzaian**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Update etcd to version 3.5.7 See release: [v3.5.7](https://github.com/etcd-io/etcd/releases/tag/v3.5.7).

**Which issue(s) this PR fixes**:
Fixes NONE #

**Does this PR introduce a user-facing change?**:
NONE

```release-note
etcd: Update to v3.5.7
```

### CODE REVIEW

**Comment by mzaian** [comment]
/hold
I created #115336 to build the image first.

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -480,7 +480,7 @@ var (
 		23: "3.5.6-0",
 		24: "3.5.6-0",
 		25: "3.5.6-0",
```
Comment (hakman) [inline cmd/kubeadm/app/constants/constants.go:482]: See https://github.com/kubernetes/kubernetes/pull/114093.
```suggestion
		22: "3.5.7-0",
		23: "3.5.7-0",
		24: "3.5.7-0",
		25: "3.5.7-0",
```

**Review by hakman — CHANGES_REQUESTED** [review]
/cc @pacoxu

File: cmd/kubeadm/app/constants/constants.go (hunk 1 above)
Comment (mzaian) [reply cmd/kubeadm/app/constants/constants.go]: Please check the changes

Comment (hakman) [reply cmd/kubeadm/app/constants/constants.go]: Why not update for k8s 1.22 also?

Comment (mzaian) [reply cmd/kubeadm/app/constants/constants.go]: Missed on that now it's done

Comment (hakman) [reply cmd/kubeadm/app/constants/constants.go]: Thanks!

**Comment by pacoxu** [comment]
https://github.com/kubernetes/k8s.io/pull/4687 is not merged. I just realized the image promotion is not completely done.

**Review by pacoxu — APPROVED** [review]

**Pull request merged**